### PR TITLE
Add deploy/destroy bash script and cleanup project

### PR DESCRIPTION
**Pull request by shumkov**

### CODE REVIEW

File: networks/mainnet.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+
+faucet_address: yhvXpqQjfN9S4j5mBKbxeGxiETJrrLETg5
```
Comment (nmarley) [inline networks/mainnet.yml]: these are not mainnet addresses...

File: lib/utils.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+function print_error() {
+   echo "$1"
+   echo ""
+   echo "$CMD_USAGE"
+   exit 1
+}
+
+function override_aws_credentials() {
+    AWS_VARS="AWS_PROFILE AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS AWS_SECRET_ACCESS AWS_REGION"
```
Comment (nmarley) [inline lib/utils.sh]: What is `AWS_SECRET_ACCESS` ? Env vars used by awscli are found here: https://docs.aws.amazon.com/cli/latest/userguide/cli-environment.html

File: lib/utils.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    for VAR_NAME in ${AWS_VARS}; do
+        FIRST_VAR_NAME="$1_$VAR_NAME"
+        SECOND_VAR_NAME="$2_$VAR_NAME";
+
+        if [[ -n ${!FIRST_VAR_NAME} ]]; then
```
Comment (nmarley) [inline lib/utils.sh]: You could just check for `$1` and `$2` at the top of the fcn & error out if both are not set.

File: .env.example [hunk 4]
```diff
@@ -1,10 +1,25 @@
-TERRAFORM_S3_BUCKET="s3-bucket-name-here"
```
Comment (nmarley) [inline .env.example]: Why remove these examples?

File: README.md [hunk 5]
```diff
@@ -1,20 +1,23 @@
-# Dash Cluster Ansible
+# Dash Networks Deployment Tool
```
Comment (nmarley) [inline README.md]: probably "Network" makes more sense

File: README.md [hunk 6]
```diff
@@ -1,20 +1,23 @@
...
 
+## Introduction
 
-## What is this?
+This tool allows deploy and manage Dash Networks networks.
```
Comment (nmarley) [inline README.md]: s/Networks networks/networks/

File: lib/utils.sh (hunk 2 above)
Comment (nmarley) [inline lib/utils.sh]: What is `AWS_SECRET_ACCESS` ? Environment variables that awscli uses are found here: https://docs.aws.amazon.com/cli/latest/userguide/cli-environment.html

File: lib/utils.sh (hunk 3 above)
Comment (nmarley) [inline lib/utils.sh]: Could probably check if `$1` and `$2` is set that the top of the function and error there instead.

File: networks/mainnet.yml (hunk 1 above)
Comment (nmarley) [inline networks/mainnet.yml]: These are not base58 encoded for mainnet.

File: .env.example (hunk 4 above)
Comment (shumkov) [reply .env.example]: It might mislead users like it's already set and they just skip configuration. And then quite complicated to find a mistake. If it's empty it's more obvious what is the problem. I faced that before.

Comment (shumkov) [reply .env.example]: I think it's better to add comments with examples if needed

File: README.md (hunk 5 above)
Comment (shumkov) [reply README.md]: I thought if we are deploying multiple networks we should keep it plural. No?
If I'm wrong we should rename repo too.

File: .env.example (hunk 4 above)
Comment (nmarley) [reply .env.example]: Ok, I take your point. Maybe we can specify instructions (link to README if not set) like you did somewhere else in this project.

File: README.md (hunk 5 above)
Comment (nmarley) [reply README.md]: It's per-network. Makes more sense in English to call it a "network deployment tool" -- it is understood that the tool can be used over & over. But you can call it what you want.

Comment (shumkov) [reply README.md]: Ok. I see. Then I will fix it everywhere.

File: .env.example (hunk 4 above)
Comment (shumkov) [reply .env.example]: Done

File: lib/terraform.sh [hunk 7]
```diff
@@ -12,10 +12,11 @@ function run_terraform_command() {
     # Initialize terraform w/remote AWS backend configuration
 
     terraform init \
-        -backend-config="bucket=$TERRAFORM_S3_BUCKET" \
-        -backend-config="key=$TERRAFORM_S3_KEY" \
+        -backend="$TERRAFORM_BACKEND"
```
Comment (nmarley) [inline lib/terraform.sh]: For one, I think this PR is too large and would prefer if backend setup moved to another PR.

Also, don't think this is the best way... I think we should default to "local" instead of skipping backend config altogether.

See: https://www.terraform.io/docs/commands/init.html#backend-initialization

> To skip backend configuration, use -backend=false. Note that some other init steps require an initialized backend, so it is recommended to use this flag only when the working directory was already previously initialized for a particular backend.
